### PR TITLE
Allow queries to be executed concurrently and track per-query execution duration and error count

### PR DESCRIPTION
**Pull request by putrasattvika**
Changes:
- Add `--max-open-conns` and `--max-idle-conns` options, allowing the exporter to execute metric queries concurrently if set to a number greater than 1. These limits are for each database.

  This was tested with two custom queries that executes `pg_sleep(1)`. The whole scraping duration is only ~1s, while previously it will take at least 2s for the scrape to finish.

- Add `pg_exporter_metric_query_last_duration_seconds` and `pg_exporter_metric_query_errors_total` metrics, which describes the last execution duration of metric queries and the number of query errors that have happened so far, respectively. The metrics cover custom user queries (specified with `--extend.query-path`) as well as the default metrics (`--no-disable-default-metrics` and `--no-disable-settings-metrics`).

The new metrics look like this:
```
# HELP pg_exporter_last_scrape_duration_seconds Duration of the last scrape of metrics from PostgreSQL.
# TYPE pg_exporter_last_scrape_duration_seconds gauge
pg_exporter_last_scrape_duration_seconds 1.022187453

# HELP pg_exporter_metric_query_errors_total Number of metric query execution errors
# TYPE pg_exporter_metric_query_errors_total counter
pg_exporter_metric_query_errors_total{datname="apprunner",query="pg_conns",server="localhost:5432"} 0
pg_exporter_metric_query_errors_total{datname="apprunner",query="pg_err_syntax",server="localhost:5432"} 1
pg_exporter_metric_query_errors_total{datname="apprunner",query="pg_err_timeout",server="localhost:5432"} 1
pg_exporter_metric_query_errors_total{datname="apprunner",query="pg_err_timeout2",server="localhost:5432"} 1
pg_exporter_metric_query_errors_total{datname="apprunner2",query="pg_conns",server="localhost:5432"} 0
pg_exporter_metric_query_errors_total{datname="apprunner2",query="pg_err_syntax",server="localhost:5432"} 1
pg_exporter_metric_query_errors_total{datname="apprunner2",query="pg_err_timeout",server="localhost:5432"} 1
pg_exporter_metric_query_errors_total{datname="apprunner2",query="pg_err_timeout2",server="localhost:5432"} 1

# HELP pg_exporter_metric_query_last_duration_seconds Duration of the last metric query
# TYPE pg_exporter_metric_query_last_duration_seconds gauge
pg_exporter_metric_query_last_duration_seconds{datname="apprunner",query="pg_conns",server="localhost:5432"} 0.006256037
pg_exporter_metric_query_last_duration_seconds{datname="apprunner",query="pg_err_syntax",server="localhost:5432"} 0.00366689
pg_exporter_metric_query_last_duration_seconds{datname="apprunner",query="pg_err_timeout",server="localhost:5432"} 1.000685042
pg_exporter_metric_query_last_duration_seconds{datname="apprunner",query="pg_err_timeout2",server="localhost:5432"} 1.004099791
pg_exporter_metric_query_last_duration_seconds{datname="apprunner2",query="pg_conns",server="localhost:5432"} 0.006669678
pg_exporter_metric_query_last_duration_seconds{datname="apprunner2",query="pg_err_syntax",server="localhost:5432"} 0.00349087
pg_exporter_metric_query_last_duration_seconds{datname="apprunner2",query="pg_err_timeout",server="localhost:5432"} 1.004025547
pg_exporter_metric_query_last_duration_seconds{datname="apprunner2",query="pg_err_timeout2",server="localhost:5432"} 1.001013876
```

Also note that we have four queries that took 1s to finish (across two databases), but the scrape only took 1.02s to finish.

### CODE REVIEW

File: cmd/postgres_exporter/server.go [hunk 1]
```diff
@@ -73,13 +85,32 @@ func NewServer(dsn string, opts ...ServerOpt) (*Server, error) {
 
 	level.Info(logger).Log("msg", "Established new database connection", "fingerprint", fingerprint)
 
+	serverConstLabels := prometheus.Labels{
+		serverLabelName: fingerprint,
+	}
+
+	// Figure out the database the client is connected to, which will be used as
+	// server metric label to differenentiate metrics that originate from
```
Comment (mikeadityas) [inline cmd/postgres_exporter/server.go:93]: Typo

Comment (mikeadityas) [inline cmd/postgres_exporter/server.go]: Typo. Should be `differentiate`.

**Review by mikeadityas — APPROVED** [review]
LGTM, only a minor typo.

File: cmd/postgres_exporter/server.go (hunk 1 above)
Comment (putrasattvika) [reply cmd/postgres_exporter/server.go]: Documentation has been updated

**Pull request merged**